packages/algob/src/internal/deployer.ts

```diff
@@ -224,8 +224,11 @@ export class DeployerDeployMode extends DeployerBasicMode implements Deployer {
     }
   }
 
-  async deployASA (name: string, flags: rtypes.ASADeploymentFlags): Promise<ASAInfo> {
-    if (this.loadedAsaDefs[name] === undefined) {
+  async deployASA (name: string, flags: rtypes.ASADeploymentFlags,
+    asaParams?: Partial<rtypes.ASADef>): Promise<ASAInfo> {
+    const asaDef = overrideASADef(this.accountsByName, this.loadedAsaDefs[name], asaParams);
```

Comment:
how about overwrite here ? because now ASADEF will give you last deployed ASA information

Reply:
Yes here we can use overwrite (and actually "overwrite" `loadedAsaDefs`). Currently i just return a copy.

Reply:
What I meant here is update yaml file to be overwritten by updated value. Because override doesn't make sense here.
If user wants to know what was deployed he would know by looking at asadef